lua/nvim-treesitter/locals.lua

```diff
@@ -130,13 +130,15 @@ function M.recurse_local_nodes(local_def, accumulator, full_match, last_match)
   if local_def.node then
     accumulator(local_def, local_def.node, full_match, last_match)
   else
-    for match_key, def in pairs(local_def) do
-      M.recurse_local_nodes(
-        def,
-        accumulator,
-        full_match and (full_match..'.'..match_key) or match_key,
-        match_key)
-    end
+	  if(type(local_def) == "table") then
```

Comment:
This will have already errored on line 133 since it already has assumed it's a table.

Reply:
I am not sure that `pairs` actually fails when running over something else than a table...

Reply:
> This will have already errored on line 133 since it already has assumed it's a table.

good point - I've updated the PR now

Reply:
> I am not sure that `pairs` actually fails when running over something else than a table...

It does for me:

```lua
Lua 5.4.3  Copyright (C) 1994-2021 Lua.org, PUC-Rio
> for k,d in pairs("a string") do print(k,v) end
stdin:1: bad argument #1 to 'for iterator' (table expected, got string)
stack traceback:
	[C]: in function 'next'
	stdin:1: in main chunk
	[C]: in ?
```

Reply:
Yep. Also me.